nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -193,4 +205,80 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
         notifyEvent(session, eventType, commentParent, commentDocumentModel);
     }
 
+    /**
+     * Checks if a document has comments.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasComments(CoreSession session, DocumentModel document) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Checks if a document has comments from a particular user.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasCommentsByUser(CoreSession session, DocumentModel document, String author) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
```

Comment:
i think on master version these two methods `hasComments` and `hasComments` should be `abstract` to be implemented by any new comment implementation / customers code ...

 On `10.10`  (when you will do you backport) you should keep `throw new UnsupportedOperationException();` for backward compatibility

Reply:
I do not agree, as there is no implementation for both `CommentManagerImpl` and `BridgeCommentManager`.

"backward compatibility" is also valid for migrating from `10.10` to `11.1`, if we can ease the process, let's do it.

Reply:
`CommentManagerImpl` -> will implemented to throw the `exception`
for the bridge version we should add an implementation to be able to creates / updated comments during the migration process

Reply:
I followed this https://github.com/nuxeo/nuxeo/pull/3663#discussion_r367873163

Reply:
It was the same implementation on the PR before :)

But anyway, there is **no** issue to have a default implementation on new methods added to an existing class... and I find it safer.